pkg/config/decode_test.go

```diff
@@ -0,0 +1,57 @@
+package config
+
+import (
+	"testing"
+
+	clitesting "github.com/dcos/dcos-cli/pkg/testing"
+	"github.com/pelletier/go-toml"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDecode(t *testing.T) {
+	treeMap := map[string]interface{}{
+		"core": map[string]interface{}{
+			"dcos_url":   "https://dcos.example.com",
+			"ssl_verify": "false",
+			"timeout":    15,
+			"reporting":  true,
+		},
+		"cluster": map[string]interface{}{
+			"name": "mr-cluster",
+		},
+	}
+	tree, _ := toml.TreeFromMap(treeMap)
+	conf, err := decode(tree)
+	require.NoError(t, err)
+
+	require.Equal(t, "https://dcos.example.com", conf.Core.URL)
+	require.Equal(t, true, conf.Core.SSL.Insecure())
+	require.Equal(t, 15, conf.Core.Timeout)
+	require.Equal(t, true, conf.Core.Reporting)
+	require.Equal(t, "mr-cluster", conf.Cluster.Name)
+}
+
+func TestDecodeWithEnvVars(t *testing.T) {
+	cleanup, err := clitesting.WithEnv(map[string]string{
+		"DCOS_URL":          "https://dcos-env.example.com",
+		"DCOS_CLUSTER_NAME": "nope",
+	})
+	require.NoError(t, err)
+	defer cleanup()
+
+	treeMap := map[string]interface{}{
+		"core": map[string]interface{}{
+			"dcos_url": "https://dcos.example.com", // this should get overriden by DCOS_URL
```

Comment:
Let's stick to a comment before `treeMap := map[string]interface{}{` instead of inline comments.